_posts/2022-10-17-10.0.0-release.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Apache Arrow 10.0.0 Release"
+date: "2022-10-17 00:00:00"
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+
+The Apache Arrow team is pleased to announce the 10.0.0 release. This covers
+over 3 months of development work and includes [**XXX resolved issues**][1]
+from [**YYY distinct contributors**][2]. See the Install Page to learn how to
+get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+Since the 9.0.0 release, Yanghong Zhong, Remzi Yang and Dan Harris
+have been invited to be committers.
+L.C. Hsie, Weston Pace and Raphael Taylor-Davies have joined the
+Project Management Committee (PMC).
+Thanks for your contributions and participation in the project!
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+
+## C++ notes
+
+## C# notes
+
+## Go notes
```

Comment:
@zeroshade can you add the release 10.0.0 go notes?

Reply:
```suggestion

## Go notes

* A new CI job has been added to run all of the tests with the `-asan` option using go1.18 ([ARROW-17324](https://issues.apache.org/jira/browse/ARROW-17324))
* Go now passes all integration tests on data types and IPC handling.
* The Go Arrow and Parquet packages now require go1.17+ ([ARROW-17646](https://issues.apache.org/jira/browse/ARROW-17646))

### Compute

The compute package that was importable via `github.com/apache/arrow/go/v9/arrow/compute` is now a separate module which requires go1.18+ (only the compute module, the rest of the packages still work fine under go1.17). ([ARROW-17456](https://issues.apache.org/jira/browse/ARROW-17456)).

Scalar and Vector kernel infrastructure has been implemented for performing compute operations providing the following functionality:

* casting Arrow Arrays from one type to another ([ARROW-17454](https://issues.apache.org/jira/browse/ARROW-17454))
* Using Filter and Take functions on an Arrow Array, Record Batch, or Table ([ARROW-17669](https://issues.apache.org/jira/browse/ARROW-17669))

### Arrow

* Sparse and Dense Union Arrays have been implemented along with appropriate builders and data type support including in IPC reading and writing. ([ARROW-3678](https://issues.apache.org/jira/browse/ARROW-3678), [ARROW-17276](https://issues.apache.org/jira/browse/ARROW-17276)). This includes scalar types for Dense and Sparse union ([ARROW-17390](https://issues.apache.org/jira/browse/ARROW-17390))
* LargeBinary, LargeString and LargeList arrays have been implemented for handling arrays with 64-bit offsets. This also included fixing a bug so that binary builders are correctly resettable. ([ARROW-8226](https://issues.apache.org/jira/browse/ARROW-8226), [ARROW-17275](https://issues.apache.org/jira/browse/ARROW-17275))
* Support for Decimal256 arrays has been implemented ([ARROW-10600](https://issues.apache.org/jira/browse/ARROW-10600))
* Automatic Endianness Conversion for non-native endianness is now an option for IPC streams ([ARROW-17219](https://issues.apache.org/jira/browse/ARROW-17219))
* CSV Writer now supports Timestamp, Date32 and Date64 types ([ARROW-17273](https://issues.apache.org/jira/browse/ARROW-17273))
* CSV Writer now supports custom formatting for boolean values ([ARROW-17277](https://issues.apache.org/jira/browse/ARROW-17277))
* The Go Arrow Library now provides a FlightSQL client and server implementation ([ARROW-17326](https://issues.apache.org/jira/browse/ARROW-17326)). An example server implementation is provided for a FlightSQL server using SQLite ([ARROW-17359](https://issues.apache.org/jira/browse/ARROW-17359))
* CSV Reader now supports schema type inference via NewInferringReader, along with options for specifying the type of some columns and skipping columns ([ARROW-17778](https://issues.apache.org/jira/browse/ARROW-17778))

### Parquet

* RowGroupReader.Column(index int) no longer panics if provided an invalid column index. Instead the signature has been changed to now return (PageReader, error) similar to other methods in the codebase. ([ARROW-17274](https://issues.apache.org/jira/browse/ARROW-17274))
* Bitpacking and other internal required implementations for ppc64le have been added for the Parquet library ([ARROW-17372](https://issues.apache.org/jira/browse/ARROW-17372))
* A bug has been fixed that caused inconsistent row information data from a table written by Athena ([ARROW-17453](https://issues.apache.org/jira/browse/ARROW-17453))
* Fixed a bug that caused panics when writing a Nullable List of Structs ([ARROW-17169](https://issues.apache.org/jira/browse/ARROW-17169))
* Key Value metadata in an Arrow Schema will be propagated to the Parquet file when using pqarrow even when not using StoreSchema ([ARROW-17627](https://issues.apache.org/jira/browse/ARROW-17627))
* A memory leak when using statistics on ByteArray columns has been fixed ([ARROW-17573](https://issues.apache.org/jira/browse/ARROW-17573))

```